example_packages/preprocess_cpp/src/preprocess_cpp.f90

```diff
@@ -0,0 +1,10 @@
+module preprocess_cpp
+  implicit none
+  private
+
+  public :: say_hello
+contains
+  subroutine say_hello
+    print *, "Hello, preprocess_cpp!"
+  end subroutine say_hello
```

Comment:
The current example won't test whether the preprocessing worked. Can you add the logic here that would do that?

For example, you could add an `#ifndef` block that encloses some invalid code that would break the compilation if that block wasn't correctly preprocessed:

```fortran
  subroutine say_hello
    print *, "Hello, preprocess_cpp!"
#ifndef TESTMACRO
    break me
#endif
  end subroutine say_hello
```

and `-DTESTMACRO` would need to be added to the flags for the example package build to succeed.

@awvwgk @LKedward does this approach seem appropriate to you?

Reply:
Sorry, @arteevraina for the delay getting back to you. I think we should leave the macros in the manifest for a separate PR, to keep them small. What I meant by "passing `-DTESTMACRO` to the flags" is simply that a user who builds the example package would need to pass it explicitly to fpm, e.g. `fpm build --flag "-DTESTMACRO" for the example package to successfully build.

Let's keep this PR small but complete. You'll still need to add a few tests (`test/fpm_test/test_manifest.f90` seems the appropriate file to add them to), and update the [manifest reference](https://github.com/fortran-lang/fpm-docs/blob/main/pages/spec/manifest.md) in a separate PR to fpm-docs.

Reply:
@milancurcic Sure thing.

Reply:
No, the command should not include `-cpp` because this example package would have `[preprocess.cpp]` in the manifest, so fpm would add `-cpp` to the default flags under the hood.

Yes, src/fpm_compiler.f90 is the place to add definitions to enable the C-preprocessor flag for each of the supported compilers.